anvill/python/anvill/arch.py

```diff
@@ -361,6 +369,8 @@ def register_family(self, reg_name) -> Tuple[RegisterBounds, ...]:
         return self._REG_FAMILY[self.register_name(reg_name)]
 
     def register_name(self, reg_name) -> Register:
+        self._has_avx |= any(reg in reg_name for reg in ["xmm", "ymm", "XMM", "YMM"])
```

Comment:
Change to the following. I'm not a fan of `any` over the list allocation that always needs to happen.

```python
        if reg_name.startswith("%"):
            ret_name = reg_name[1:].upper()
        else:
            ret_name = reg_name.upper()

        if ret_name.startswith("ZMM"):
            self._has_avx512 = True
        elif ret_name.startswith("YMM"):
            self._has_avx = True

        return ret_name
```

Reply:
I updated the checking for `YMM` and `XMM` registers.